train_segmentation.py

```diff
@@ -491,7 +494,12 @@ def main(params, config_path):
     num_devices = params['global']['num_gpus']
     assert num_devices is not None and num_devices >= 0, "missing mandatory num gpus parameter"
     # list of GPU devices that are available and unused. If no GPUs, returns empty list
-    lst_device_ids = get_device_ids(num_devices) if torch.cuda.is_available() else []
+    max_used_ram = get_key_def('max_used_ram', params['global'], 2000)
+    max_used_perc = get_key_def('max_used_perc', params['global'], 15)
```

Comment:
nice use of get_key_def(). You could even force an "int" as input with the [expected_type](https://github.com/NRCan/geo-deep-learning/blob/8d85078a1aeae791244807660c5cd5119cca7d83/utils/utils.py#L126) argument. Just a suggestion.